src/main/java/io/paradoxical/v2/DockerCreator.java

```diff
@@ -63,7 +62,14 @@ public static Container build(
         }
 
         for (Integer transientPort : config.getTransientPorts()) {
-            ports.bind(ExposedPort.tcp(transientPort), Ports.Binding.bindPort(random.nextInt(30000) + 15000));
+            int port = random.nextInt(30000) + 15000;
+            try {
+                ServerSocket serverSocket = new ServerSocket(0);
```

Comment:
Passing in 0 here gets us an available port: https://docs.oracle.com/javase/7/docs/api/java/net/ServerSocket.html#ServerSocket(int)